libs/content-search-toolkit/src/services/utils/utils.spec.ts

```diff
@@ -0,0 +1,19 @@
+import { filterDoc } from './index'
+
+describe('filterDoc', () => {
+  it('should prune fields that are longer than the letter limit', () => {
+    const test: Record<string, unknown> = {
+      a: 'some-value',
+      b: {
+        c: 'some-really-long-value-that-should-get-pruned',
+      },
+    }
+
+    // Also make the test object circular
+    test.d = test
```

Comment:
Circularity should be another, separate, test case

Reply:
Agreed, I made another, very simple test for that